client/state/comments/reducer.js

```diff
@@ -127,6 +128,39 @@ export const fetchStatusInitialState = {
 	hasReceivedAfter: false,
 };
 
+const isValidExpansionsAction = action => {
+	const { siteId, postId, commentIds, displayType } = action.payload;
+	return (
+		siteId &&
+		postId &&
+		isArray( commentIds ) &&
+		includes( values( POST_COMMENT_DISPLAY_TYPES ), displayType )
+	);
+};
+
+export const expansions = createReducer(
+	{},
+	{
+		[ READER_EXPAND_COMMENTS ]: ( state, action ) => {
+			const { siteId, postId, commentIds, displayType } = action.payload;
+
+			if ( ! isValidExpansionsAction( action ) ) {
+				return state;
+			}
+
+			const stateKey = getStateKey( siteId, postId );
+
+			// generate object of { [ commentId ]: displayType }
+			const newVal = zipObject( commentIds, fill( Array( commentIds.length ), displayType ) );
```

Comment:
I just learned about `zipObject`! It was instead of doing:
```javascript
const newVal = {};
commentIds.forEach( x => newVal[ x ] = displayType );
```

picking fp methods instead of faster mutable ones sometimes feels weird 🤷‍♂️